homeassistant/core.py

```diff
@@ -855,14 +855,11 @@ def __init__(self, hass: HomeAssistant) -> None:
         self._hass = hass
         self._async_unsub_call_event = None  # type: Optional[CALLBACK_TYPE]
 
-        def _gen_unique_id() -> Iterator[str]:
-            cur_id = 1
-            while True:
-                yield '{}-{}'.format(id(self), cur_id)
-                cur_id += 1
-
-        gen = _gen_unique_id()
-        self._generate_unique_id = lambda: next(gen)
+    def _generate_unique_id(self) -> Iterator[str]:
```

Comment:
This is not the same.

Reply:
Lambdas can't be typed and mypy doesn't always automatically can infer type.

The difference I see is that previously if there were several instances of `ServiceRegistry` - each would get their own IDs. Now the IDs are globally unique.

Any other difference?

Reply:
Calling this function now returns an iterator. In the old approach it would wrap that generator function in a lambda calling `next` on the return value.

Reply:
removing that function in #15674

Reply:
Merged